node/src/commands/genesis/mod.rs

```diff
@@ -0,0 +1,270 @@
+    force: &bool,
+) -> Result<()> {
+    let inputs_path = Path::new(inputs_path);
+    let output_path = Path::new(output_path);
+
+    if !force {
+        if let Ok(file_exists) = output_path.try_exists() {
+            if file_exists {
+                return Err(anyhow!("Failed to generate new genesis file {} because it already exists. Use the --force flag to overwrite.", output_path.display()));
+            }
+        } else {
+            return Err(anyhow!("Failed to open {} file.", output_path.display()));
+        }
+    }
+
+    if let Ok(file_exists) = inputs_path.try_exists() {
+        if !file_exists {
+            return Err(anyhow!(
+                "The {} file does not exist. It is necessary to generate the genesis file",
+                inputs_path.display()
+            ));
+        }
+    } else {
+        return Err(anyhow!("Failed to open {} file.", inputs_path.display()));
+    }
+
+    let parent_path = match output_path.parent() {
+        Some(path) => path,
+        None => {
+            return Err(anyhow!(
+                "There has been an error processing output_path: {}",
+                output_path.display()
+            ))
+        },
+    };
+
+    let genesis_input: GenesisInput = load_config(inputs_path).extract().map_err(|err| {
+        anyhow!("Failed to load {} genesis input file: {err}", inputs_path.display())
+    })?;
+    println!("Genesis input file: {} has successfully been loaded.", output_path.display());
+
+    let accounts = create_accounts(&genesis_input.accounts, parent_path, force)?;
+    println!(
+        "Accounts have successfully been created at: {}/{}",
+        parent_path.display(),
+        DEFAULT_ACCOUNTS_DIR
+    );
+
+    let genesis_state = GenesisState::new(accounts, genesis_input.version, genesis_input.timestamp);
+    fs::write(output_path, genesis_state.to_bytes()).unwrap_or_else(|_| {
+        panic!("Failed to write genesis state to output file {}", output_path.display())
+    });
+    println!("Miden node genesis successful: {} has been created", output_path.display());
+
+    Ok(())
+}
+
+/// Converts the provided list of account inputs into [Account] objects.
+///
+/// This function also writes the account data files into the default accounts directory.
+fn create_accounts(
+    accounts: &[AccountInput],
+    parent_path: &Path,
+    force: &bool,
+) -> Result<Vec<Account>> {
+    let mut accounts_path = PathBuf::from(&parent_path);
+    accounts_path.push(DEFAULT_ACCOUNTS_DIR);
+
+    let accounts_exists = accounts_path.try_exists()?;
+
+    if accounts_exists {
+        if *force {
+            fs::remove_dir_all(&accounts_path)?;
+        } else {
+            return Err(anyhow!("Failed to generate accounts folder {} because it already exists. Use the --force flag to overwrite.", accounts_path.display()));
+        }
+    }
+
+    fs::create_dir_all(&accounts_path)
+        .map_err(|err| anyhow!("Failed to create accounts directory: {err}"))?;
```

Comment:
Another option to do this would be:

1. Do not delete the directory here.
2. If the directory does not exist, create it.
3. When writing individual account files (on line 169 below), throw an error if a file exists and `force` flag is not on.

The benefit of this is that we never delete the whole directory (which may contain files we are not planning to overwrite).